app/controllers/users_controller.rb

```diff
@@ -51,7 +51,8 @@ def transfer
       confirm_change(@user, "Added #{@user.full_name} to roster.")
       redirect_to roster_users_path(@roster)
     else
-      report_errors(@user, fallback_location: roster_users_path)
+      flash.now[:errors] = @user.errors.full_messages
+      render :index, status: :unprocessable_entity
```

Comment:
This one is weird since it's a single-field form on the index and doesn't have a dedicated "form" page. Meaning it's actually more appropriate to keep this as a redirect (but hardcoded to go back to the index)